src/view_neume.cpp

```diff
@@ -93,249 +97,176 @@ void View::DrawNc(DeviceContext *dc, LayerElement *element, Layer *layer, Staff
-                DrawNc(dc, dynamic_cast<LayerElement *>(ncVector.at(3)), layer, staff, measure, SMUFL_E990_chantPunctum, noteWidth, 0);
+            if (pitchDifferences.at(1) == 1) {
+                if (position == 1) {
+                    params.at(0).fontNo = SMUFL_E9B9_chantLigaturaDesc2nd;
+                }
+                else if (position == 2) {
+                    params.at(0).fontNo = SMUFL_E9BD_chantConnectingLineAsc2nd;
+                }
             }
-            else {
-                DrawNc(dc, dynamic_cast<LayerElement *>(ncVector.at(1)), layer, staff, measure, SMUFL_E990_chantPunctum, noteHeight/4, 0);
-                DrawNc(dc, dynamic_cast<LayerElement *>(ncVector.at(2)), layer, staff, measure, lineCode);
-                DrawNc(dc, dynamic_cast<LayerElement *>(ncVector.at(2)), layer, staff, measure, SMUFL_E990_chantPunctum);
-                DrawNc(dc, dynamic_cast<LayerElement *>(ncVector.at(3)), layer, staff, measure);
+            else if (pitchDifferences.at(1) == 2) {
+                if (position == 1) {
+                    params.at(0).fontNo = SMUFL_E9BA_chantLigaturaDesc3rd;
+                }
+                else if (position == 2) {
+                    params.at(0).fontNo = SMUFL_E9BD_chantConnectingLineAsc2nd;
+                }
             }
             break;
         }
         case PRESSUS:
         {
-            // Two punctums followed by a punctum inclinatum
-            // No additional logic required
-            
-            DrawNc(dc, dynamic_cast<LayerElement *>(ncVector.at(0)), layer, staff, measure);
-            DrawNc(dc, dynamic_cast<LayerElement *>(ncVector.at(1)), layer, staff, measure);
-            DrawNc(dc, dynamic_cast<LayerElement *>(ncVector.at(2)), layer, staff, measure,
-                    SMUFL_E991_chantPunctumInclinatum);
+            params.at(0).fontNo = (position == 2) ? SMUFL_E991_chantPunctumInclinatum : SMUFL_E990_chantPunctum;
             break;
         }
-        default: DrawLayerChildren(dc, neume, layer, staff, measure);
+        default: break;
+    } 
+    // If the nc is supposed to be a virga and currently is being rendered as a punctum
+    // change it to a virga
+    if (nc->HasAttribute("diagonalright", "u") && params.at(0).fontNo == SMUFL_E990_chantPunctum) {
+        params.at(0).fontNo = SMUFL_E996_chantPunctumVirga;
     }
 
+    const int noteHeight = (int)(m_doc->GetDrawingDoubleUnit(staff->m_drawingStaffSize) / 2);
+    const int noteWidth = (int)(m_doc->GetDrawingDoubleUnit(staff->m_drawingStaffSize) / 1.4);
+    
+    for (auto it = params.begin(); it != params.end(); it++) {
+        DrawSmuflCode(dc, noteX + it->xOffset * noteWidth, yValue + it->yOffset * noteHeight,
+               it->fontNo, staff->m_drawingStaffSize, false, true);
+    } 
+    
+    // Draw the children
+    DrawLayerChildren(dc, nc, layer, staff, measure);

```

Comment:
Can Nc have a child? If not you can remove it.

Reply:
As of now I don't think so but it might when MEI 4 is finalized.

Reply:
OK, leave it then.